bundles/org.openhab.binding.panamaxfurman/src/main/java/org/openhab/binding/panamaxfurman/internal/transport/PanamaxFurmanSocket.java

```diff
@@ -0,0 +1,211 @@
+            try {
+                telnetSocket = new Socket();
+                telnetSocket.connect(new InetSocketAddress(address, port), CONNECT_TIMEOUT_MILLIS);
+                telnetSocket.setKeepAlive(true);
+                // Block forever on reads
+                telnetSocket.setSoTimeout(INFINITE_TIMEOUT);
+                InputStream is = telnetSocket.getInputStream();
+                OutputStream os = telnetSocket.getOutputStream();
+                this.inputStreamReaderThread = new InputStreamReaderThread(is);
+                this.inputStreamReaderThread.start();
+                logger.trace("Started reader thread, pausing   @{}", connectionName);
+                Thread.sleep(500);
+                logger.trace("Pause complete   @{}", connectionName);
+                this.writer = new BufferedWriter(new OutputStreamWriter(os, StandardCharsets.US_ASCII));
+                logger.debug("Connected   @{}", connectionName);
+            } catch (IOException | InterruptedException e) {
+                // should any errors occur, clean everything up
+                close();
+                throw e;
+            }
+        }
+    }
+
+    @SuppressWarnings("null")
+    protected void send(@Nullable String textToSend) throws IOException {
+        synchronized (ioMutex) {
+            String command = textToSend;
+            try {
+                writer.write(command + "\r");
+                writer.flush();
+                logger.debug("sent: {}    @{}", command, connectionName);
+            } catch (IOException e) {
+                close();
+                throw e;
+            }
+        }
+    }
+
+    @SuppressWarnings("null")
+    protected void close() {
+        synchronized (ioMutex) {
+            if (inputStreamReaderThread != null) {
+                inputStreamReaderThread.shutdown();
+            }
+            Util.closeQuietly(writer);
+            Util.closeQuietly(telnetSocket);
+        }
+    }
+
+    /**
+     * Read incoming data from the Power Conditioner device and notify listeners accordingly.
+     *
+     * @author Dave Badia - Initial contribution
+     *
+     */
+    private class InputStreamReaderThread extends Thread {
+        private volatile boolean stopReader = false;
+        private BufferedReader bufferedReader;
+
+        /**
+         * This latch is used to track when the thread has terminated
+         */
+        private CountDownLatch stopLatch = new CountDownLatch(1);
+
+        /**
+         * Construct a reader thread to poll the given InputStream for incoming data
+         */
+        private InputStreamReaderThread(InputStream inputStream) {
+            logger.trace("creating InputStreamReaderThread  @{}", connectionName);
+            this.bufferedReader = new BufferedReader(new InputStreamReader(inputStream));
+
+            this.setDaemon(true);
+            this.setName("PanamaxFurmanInputStreamReader-" + connectionName);
+        }
+
+        @SuppressWarnings("null")
```

Comment:
null suppression should be avoided, please handle them in code.

Reply:
Yep, I already do but eclipse gives me "Potential null access..." warnings even though I'm checking for null.  Agreed they aren't/should not be necessary though, I will remove all of them